app/main/views/g_cloud.py

```diff
@@ -595,10 +595,57 @@ def did_you_award_contract(framework_framework, project_id):
 
 @direct_award.route(
     '/<string:framework_framework>/projects/<int:project_id>/which-service-won-contract',
-    methods=['GET']
+    methods=['GET', 'POST']
 )
 def which_service_won_contract(framework_framework, project_id):
-    abort(404)
+    project = data_api_client.get_direct_award_project(project_id=project_id)[
+        'project']
+
+    if not is_direct_award_project_accessible(project, current_user.id):
+        abort(404)
+    if not project['lockedAt']:
+        abort(400)
+
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(
+        all_frameworks, framework_framework)
+
+    search = data_api_client.find_direct_award_project_searches(user_id=current_user.id,
+                                                                project_id=project['id'],
+                                                                only_active=True)['searches'][0]
+
+    framework_slug = search_api_client.get_index_from_search_api_url(
+        search['searchUrl'])
+    download_results_manifest = content_loader.get_manifest(
+        framework_slug, 'download_results')
+    required_fields = get_fields_from_manifest(download_results_manifest)
+
+    services = data_api_client.find_direct_award_project_services(
+        project_id=project['id'],
+        user_id=current_user.id,
+        fields=required_fields,
+    )
+
+    form = WhichServiceWonTheContractForm(services)
+
+    if request.method == "POST" and form.validate_on_submit():
+        flash('Contract awarded.')
+        return redirect(url_for('.view_project', framework_framework=framework_framework, project_id=project['id']))
```

Comment:
Why does this redirect to `.view_project`? Was the flash message part of the prototype?

Reply:
no but currently there's no route for me to hit, once your stuff is, it will go there.